### PR TITLE
ci: automate merging of Crowdin PR before publishing

### PR DESCRIPTION
**Pull request by matchai**
#### Description

Locate and merge the existing Crowdin PR as part of the deployment process.

#### Motivation and Context

Merging existing Crowdin PRs is currently a manual process that is often forgotten before a release.

### CODE REVIEW

**Comment by davidkna** [comment]
I think just running `gh pr merge --squash i18n_master` would also work.

**Comment by matchai** [comment]
Much better! 😄

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -272,11 +272,22 @@ jobs:
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  publish_docs:
+    name: Publish docs to Netlify
+    runs-on: ubuntu-latest
+    needs: merge_crowdin_pr
     steps:
       - name: Setup | Checkout
         uses: actions/checkout@v3
```
Comment (davidkna) [inline .github/workflows/deploy.yml:293]: I think this needs `with: ref: master` to include the just-merged commit.

Comment (matchai) [reply .github/workflows/deploy.yml:293]: Yep! Excellent point! 👍

Comment (davidkna) [reply .github/workflows/deploy.yml:293]: And I just noticed `notarize_and_pkgbuild` will also build the docs. Maybe all deploy workflows should use it?

Comment (matchai) [reply .github/workflows/deploy.yml:297]: I've gone ahead and added the ref and dependency on the `merge_crowdin_pr` step. I think that should be fine for now 👍

**Review by davidkna — APPROVED** [review]

**Pull request merged**